src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -12,76 +23,78 @@
   }
 
   @Override
   // get a specific user. return null if not found. TODO: User Validation
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userString = request.getParameter("user");
-
-    PersonObject result =
-        people.stream()
-            .filter(person -> userString.equals(person.getGitHubID()))
-            .findAny()
-            .orElse(null);
-    if (result == null) {
+    String userGithub = request.getParameter("githubID");
+    Key userKey = KeyFactory.createKey(PersonObject.ENTITY_NAME, userGithub);
+
+    PersonObject userObject;
+
+    try {
+      userObject = toPersonObject(datastore.get(userKey));
+    } catch (Exception e) {
       sendRawTextError(
-          response, HttpServletResponse.SC_BAD_REQUEST, "invalid user or user does not exist");
+          response, HttpServletResponse.SC_BAD_REQUEST, "Person does not exist in Datastore");
+      userObject = null;
     }
 
-    String jsonPerson = new Gson().toJson(result);
+    String jsonPerson = new Gson().toJson(userObject);
     response.setContentType("application/json;");
     response.getWriter().println(jsonPerson);
   }
 
+  @Override
+  // get a specific user. return null if not found.
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userGitHubID = request.getParameter("gitHubID");
+    if (userGitHubID.isBlank()) {
+      sendRawTextError(response, HttpServletResponse.SC_BAD_REQUEST, "Invalid ID");
+      return;
+    }
+    ArrayList<String> userTags =
+        (ArrayList<String>) Arrays.asList(request.getParameterValues("tags"));
+
+    sendData(new PersonBuilder().gitHubID(userGitHubID).interestTags(userTags).buildPerson());
+  }
+
+  public void sendData(PersonObject person) {
+    Entity personEntity = new Entity(PersonObject.ENTITY_NAME, person.getGitHubID());
+    personEntity.setProperty(PersonObject.GITHUB_ID_FIELD, person.getGitHubID());
+    personEntity.setProperty(PersonObject.TAG_LIST_FIELD, person.getTags());
+    datastore.put(personEntity);
+  }
+
+  public PreparedQuery queryInDatabase(
+      Entity EntityName, String field, Object ThingToBeCompareTo, FilterOperator comparator) {
+
+    Filter userFilter = new FilterPredicate(field, comparator, ThingToBeCompareTo);
+
+    return datastore.prepare(new Query(PersonObject.ENTITY_NAME).setFilter(userFilter));
+  }
+
+  public PersonObject toPersonObject(Entity personEntity) {
+    PersonBuilder userBuilder = new PersonBuilder();
+
+    String entityGitHubID = (String) personEntity.getProperty(PersonObject.GITHUB_ID_FIELD);
+    userBuilder.gitHubID(entityGitHubID);
+    ArrayList<String> entityTagList =
+        (ArrayList<String>)
+            Arrays.asList(personEntity.getProperty(PersonObject.TAG_LIST_FIELD)).stream()
+                .map(tag -> (String) tag)
+                .collect(Collectors.toList());
```

Comment:
Q: Is this to avoid the unsafe casting warning from doing something like:
```java
ArrayList<String> entityTagList = (ArrayList<String>)personEntity.getProperty(PersonObject.TAG_LIST_FIELD);
```

Reply:
Yeah, It seemed like good practice to have the explicit casting there

Reply:
Is there a better way to do this?

Reply:
This looks ok to me!